proxyd/latest_block_poller.go

```diff
@@ -0,0 +1,78 @@
+package proxyd
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type LatestBlockPoller struct {
+	rt       RoundTripper
+	bn       atomic.Uint64
+	shutdown bool
+	mutex    sync.Mutex
+}
+
+type RoundTripper func(ctx context.Context, req json.RawMessage) (*RPCRes, error)
+
+// NewLatestBlockPoller creates a new LatestBlockPoller and starts polling
+// for the latest block number in a separate goroutine.
+func NewLatestBlockPoller(rt RoundTripper) *LatestBlockPoller {
+	p := &LatestBlockPoller{
+		rt: rt,
+	}
+	p.poll()
+	go p.start()
```

Comment:
The comment below addressing the `start` method would allow you to remove this additional `p.poll()` call.

Reply:
done